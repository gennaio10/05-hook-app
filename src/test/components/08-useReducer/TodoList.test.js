import React from "react";
import "@testing-library/jest-dom";
import { demoTodos } from "../../fixtures/demoTodos";
import { shallow } from "enzyme";
import { TodoList } from "../../../components/08-useReducer/TodoList";

describe("Pruebas en <TodoList />", () => {
  const handleToggle = jest.fn();
  const handleDelete = jest.fn();

  const wrapper = shallow(
    <TodoList
      todos={demoTodos}
      handleToggle={handleToggle}
      handleDelete={handleDelete}
    />
  );

  test("debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de tener los elementos de demoTodos", () => {
    expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length)
  });

  test("debe de tener la funcion handleDelete", () => {
    expect(wrapper.find('TodoListItem').at(0).prop('handleDelete'))
    .toEqual(expect.any(Function))
  });
});
