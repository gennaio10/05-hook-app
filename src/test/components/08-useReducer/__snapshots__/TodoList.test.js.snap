// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoList /> debe de mostrarse correctamente 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={0}
    key="1"
    todo={
      Object {
        "desc": "Aprender Mongo",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={1}
    key="2"
    todo={
      Object {
        "desc": "Aprender Express",
        "done": false,
        "id": 2,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={2}
    key="3"
    todo={
      Object {
        "desc": "Aprender React",
        "done": false,
        "id": 3,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={3}
    key="4"
    todo={
      Object {
        "desc": "Aprender Node",
        "done": false,
        "id": 4,
      }
    }
  />
</ul>
`;
